lib/_stream_readable.js

```diff
@@ -538,12 +538,12 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   }
   // This is a brutally ugly hack to make sure that our error handler
   // is attached before any userland ones.  NEVER DO THIS.
-  if (!dest._events || !dest._events.error)
+  if (!dest._events || !dest._events.has('error'))
     dest.on('error', onerror);
-  else if (Array.isArray(dest._events.error))
-    dest._events.error.unshift(onerror);
+  else if (Array.isArray(dest._events.get('error')))
+    dest._events.get('error').unshift(onerror);
   else
-    dest._events.error = [onerror, dest._events.error];
+    dest._events.set('error', [onerror, dest._events.get('error')]);
```

Comment:
I believe all of these changes to readable stream are going to result in the same problem as when I proposed optimizations for `eventEmitter.once()`, which is that some modules on npm have their `readable-stream` pinned to specific versions or limited version ranges. So if someone uses an older `readable-stream` on an io.js with these changes, things will break.